src/transformers/models/llama/modeling_llama.py

```diff
@@ -283,23 +283,37 @@ def __init__(self, dim, max_position_embeddings=2048, base=10000, device=None, s
         super().__init__(dim, max_position_embeddings, base, device)
 
     def _set_cos_sin_cache(self, seq_len, device, dtype):
+        if seq_len <= self.max_position_embeddings:
+            seq_len = self.max_position_embeddings
+
         self.max_seq_len_cached = seq_len
 
         if seq_len > self.max_position_embeddings:
             base = self.base * (
                 (self.scaling_factor * seq_len / self.max_position_embeddings) - (self.scaling_factor - 1)
             ) ** (self.dim / (self.dim - 2))
-            inv_freq = 1.0 / (base ** (torch.arange(0, self.dim, 2).float().to(device) / self.dim))
-            self.register_buffer("inv_freq", inv_freq, persistent=False)
+        else:
+            base = self.base
+        inv_freq = 1.0 / (base ** (torch.arange(0, self.dim, 2).float().to(device) / self.dim))
 
-        t = torch.arange(self.max_seq_len_cached, device=device, dtype=self.inv_freq.dtype)
+        t = torch.arange(self.max_seq_len_cached, device=device, dtype=inv_freq.dtype)
 
-        freqs = torch.einsum("i,j->ij", t, self.inv_freq)
+        freqs = torch.einsum("i,j->ij", t, inv_freq)
         # Different from paper, but it uses a different permutation in order to obtain the same calculation
         emb = torch.cat((freqs, freqs), dim=-1)
         self.register_buffer("cos_cached", emb.cos().to(dtype), persistent=False)
         self.register_buffer("sin_cached", emb.sin().to(dtype), persistent=False)
 
+    def forward(self, x, seq_len=None):
+        # x: [bs, num_attention_heads, seq_len, head_size]
+        if not (seq_len <= self.max_seq_len_cached == self.max_position_embeddings):
```

Comment:
As it is written, this condition is `True` when:
1. seq_len > self.max_seq_len_cached OR
2. self.max_seq_len_cached != self.max_position_embeddings

This means that this is always called except when `seq_len == self.max_seq_len_cached == self.max_position_embeddings`, resulting in a big slowdown.

What we want here is probably 
```
if seq_len > self.max_seq_len_cached or (seq_len < self.max_seq_len_cached and self.max_seq_len_cached > self.max_position_embeddings):
```
This translates to `True` strictly when we need to recompute the sin cos variables, saving precious time!